packages/@aws-cdk/aws-appmesh/lib/gateway-route-spec.ts

```diff
@@ -1,32 +1,408 @@
+   */
+  DISABLED = 'DISABLED',
+}
+
+/**
+ * All properties for gRPC gateway route rewrite.
+ */
+export interface HttpGatewayRouteRewriteConfig {
+  /**
+   * GatewayRoute CFN configuration for gRPC gateway route host name rewrite.
+   *
+   * @default - rewrite hostname to the destination Virtual Service name.
+   */
+  readonly hostname?: CfnGatewayRoute.GatewayRouteHostnameRewriteProperty;
+
+  /**
+   * GatewayRoute CFN configuration for gRPC gateway route path rewrite.
+   *
+   * @default - rewrite path to '/'.
+   */
+  readonly path?: CfnGatewayRoute.HttpGatewayRoutePathRewriteProperty;
+
+  /**
+   * GatewayRoute CFN configuration for gRPC gateway route prefix rewrite.
+   *
+   * @default - rewrite prefix to '/'.
+   */
+  readonly prefix?: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty;
+}
+
+/**
+ * Defines gRPC gateway route rewrite that can be defined in addition to host name.
+ */
+export abstract class HttpGatewayRoutePathOrPrefixRewrite {
+  /**
+   * The path to rewrite.
+   *
+   * @param exact The exact path to rewrite.
+   */
+  public static path(exact: string): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePathRewriteImpl(exact);
+  }
+
+  /**
+   * The default prefix used to replace the incoming route prefix when rewritten.
+   *
+   * @param isDefault Enable or disable the default prefix
+   */
+  public static defaultPrefix(isDefault: Default): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePrefixRewriteImpl({ defaultPrefix: isDefault } );
+  }
+
+  /**
+   * Replace the incoming route prefix when rewritten.
+   *
+   * @param value The value used to replace the incoming route prefix when rewritten.
+   */
+  public static customPrefix(value: string): HttpGatewayRoutePathOrPrefixRewrite {
+    return new HttpGatewayRoutePrefixRewriteImpl({ value: value } );
+  }
+
+  /**
+   * Return HTTP gateway route rewrite configuration.
+   */
+  abstract bind(scope: Construct): HttpGatewayRouteRewriteConfig;
+}
+
+class HttpGatewayRoutePathRewriteImpl extends HttpGatewayRoutePathOrPrefixRewrite {
+  constructor(
+    private readonly exact: string,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRouteRewriteConfig {
+    return {
+      path: {
+        exact: this.exact,
+      },
+    };
+  }
+}
+
+class HttpGatewayRoutePrefixRewriteImpl extends HttpGatewayRoutePathOrPrefixRewrite {
+  constructor(
+    private readonly prefixRewrite: CfnGatewayRoute.HttpGatewayRoutePrefixRewriteProperty,
+  ) { super(); }
+
+  bind(_scope: Construct): HttpGatewayRouteRewriteConfig {
+    return {
+      prefix: this.prefixRewrite,
+    };
+  }
+}
+
+/**
+ * The criterion for determining a rewrite for this HTTP GatewayRoute.
+ */
+export interface HttpGatewayRouteRewrite {
+  /**
+   * The default target host name to write to.
+   *
+   * @default - rewrite hostname to the destination Virtual Service name.
+   */
+  readonly defaultHostname?: Default;
```

Comment:
Is there any reason we can't make this a boolean instead of an enum?

Reply:
No, I can design this to accept `boolean` and use ternary to assign `ENABLED` or `DISABLED`

```ts
public static defaultPrefix(isDefault: boolean): HttpGatewayRoutePathOrPrefixRewrite {
    return new HttpGatewayRoutePrefixRewriteImpl({ defaultPrefix: isDefault ? 'ENABLED' : 'DISABLED' } );
  }
```